pkg/network/network.go

```diff
@@ -136,6 +140,75 @@ func Render(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, opensh
 	return objs, nil
 }
 
+// RenderObjsToRemove creates removal component object list
+func RenderObjsToRemove(prev, conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, openshiftNetworkConfig *osv1.Network, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
+	log.Print("starting rendering objects to delete phase")
+	objsToRemove := []*unstructured.Unstructured{}
+	o := []*unstructured.Unstructured{}
+	var err error
+
+	operatorNamespace := os.Getenv("OPERATOR_NAMESPACE")
+
+	if conf.Multus == nil && prev != nil {
+		// render Multus
+		o, err = renderMultus(prev, manifestDir, openshiftNetworkConfig, clusterInfo)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	if conf.LinuxBridge == nil && prev != nil {
+		// render LinuxBridge for Removal
+		o, err = renderLinuxBridge(prev, manifestDir, clusterInfo)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	if conf.KubeMacPool == nil && prev != nil {
+		// render kubeMacPool
+		o, err = renderKubeMacPool(prev, manifestDir)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	if conf.NMState == nil && prev != nil {
+		// render NMState for Removal
+		o, err = renderNMState(prev, manifestDir, clusterInfo)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	if conf.Ovs == nil && prev != nil {
+		// render Ovs for Removal
+		o, err = renderOvs(prev, manifestDir, clusterInfo)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	// Remove Namespace occurences, since its not allowed to remove it, as cnao pod is there
```

Comment:
I am not sure why are we checking for cnao namespace. is removing the cnao instance a possible scenario that requires protecting? Please elaborate

Reply:
part of the module rendering has the namespace where the cnao pod itself resides
so its forbidden to remove it, else you will end without cnao operator

Reply:
I understand the motive, but not sure it is plausible - the objsToRemove list is comprised by very specific modules :ovs, nmstate, kmp, linux-bridge and multus. 
any object in objsToRemove, no matter its namespace, could not be the cnao pod. could it?

Reply:
the object will be a list of k8s entities such as namespace, deamonset , configmap, clusterrole, clusterRolebinding, 

all need to be deleted beside the namespace, the namespace was generated as part of the creation and since the component is now at the operator component , the ns cant be deleted, it was tested and groomed with Petr as well
on the demo i will show the parts that are removed

Reply:
The code is not as clear as it could be but the situation is following:

All components are created in OPERAND_NAMESPACE, operator is running in OPERATOR_NAMESPACE. By default, these variables are the same. We need to make sure we don't ever remove OPERATOR_NAMESPACE, OPERAND_NAMESPACE can be removed only if it is not used anymore. I'm fine with leaving the removal of it and just keep it hanging no matter what. We can handle it in a follow-up.

Reply:
so any ai or not as i understand ?

Reply:
There is an action item. For now let's settle with "We will not remove any namespace". So iterate objsToRemove and drop all objects of Namespace type. Add a TODO comment to cleanup OPERAND_NAMESPACE once there are no components using it.

Reply:
please see if i did fine, thanks